### ISSUE TITLE
Externals: some external libraries (ab)use

### ISSUE THREAD

**Issue by szaghi** [Fortran-FOSS-Programmers/WenOOF#5]
Hi all,

I am planning to insert some external libraries:
- a module for kinds parametrization and portability and _numbers auxiliary/helper procedures_ (essentially for casting numbers to strings and viceversa); I will start from a reduced version of [IR_Precision](https://github.com/szaghi/IR_Precision), however I would like to have your suggestions on other possibilities;
- for the test suite it is useful to have an _odepack_ in oop; I am planning to start a new small library based on the hints of Jacob [oop-runge-kutta](http://degenerateconic.com/object-oriented-runge-kutta-module/), however feel free to suggest other options;
- a plotting library is useful for the rapid check of results; gnuplot is great, but I would like to play with other solutions... like the one of Jacob https://github.com/jacobwilliams/pyplot-fortran

What do think?

**Comment by jacobwilliams** [comment]
Note: I have a slightly expanded version of this RK module [here](https://github.com/jacobwilliams/Fortran-Astrodynamics-Toolkit/blob/master/src/rk_module.f90). Feel free to use it (BSD license) or just to get some inspiration.  Do you intend to start a new project for this, or have it be included in this one?  I certainly think that a good comprehensive object-oriented integration package is needed (with selectable methods, fixed and variable-step, event finding, ...), and could be an entirely separate project.

**Comment by szaghi** [comment]
Hi @jacobwilliams ,

concerning the _odepack_ oop, I hope we can start a new separate project with the features you listed a many others. If anyone will not open a new project tomorrow I will open it starting from your library that is a very good init, all are adviced :-)

See you soon.

**Comment by zbeekman** [comment]
@szaghi I have a question: From your earlier issue, #1 , it sounds like WenOOF will be strictly a WENO interpolation library, without any knowledge of PDEs, flux computations etc., so will the time integrator routines/module be used just for the test suite? Or will it be a service offered/exposed by the WenOOF library interface?

My second point is on the topic of time integrators: It would be wonderful to have an object oriented library of time integrators that could be used, for integration of abstract data type (ADT) calculus (See Rouson et al.) with procedures to query their properties, like their [stability locus plots](https://en.wikipedia.org/wiki/Stiff_equation#A-stability) sigma roots, etc. Users could then be guided to picking a good time integrator for their PDE/ODE of interest and it would take a lot of the headache out of picking a suitable time integration scheme to match your spatial scheme when making a PDE solver.

**Comment by szaghi** [comment]
@zbeekman 

Yes, the _odepack_ oop will be used in Wenoof only into the testing suite. I like to have it only for my bacground, I hope that other Fortraners with different background implement other tests not related to PDEs.

The odepack will be a standalone project with all the features we want. I am greately inspired by the work of @jacobwilliams and by the Rouson et. al book.

**Comment by jacobwilliams** [comment]
Just don't call it "ODEPACK", because [that](http://www.netlib.org/odepack/) already exists.  :)

**Comment by szaghi** [comment]
:-)   mmm, as someone already said, the code names I found are terrific... in Italian the only names coming to me is _FrODE_, Fortran ODE integration library, the acronym in Italian means _fraud_...

**Comment by szaghi** [comment]
Hi all,

I have added IR_Precision, see last post of issue#3.

If _all_ agree, I am planning to start a parallel project implementing an ODE-solving library in modern OOP. I need it for the Riemann-solver tester. Can I start?
